Rename createImageBitMaps's imageOrientation from "none" to "from-image"

yiyix opened the pull request:
- [x] At least two implementers are interested (and none opposed):
   * Chrome
   * Firefox (https://github.com/whatwg/html/issues/8085#issue-1297039069)
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/web-platform-tests/wpt/pull/37880 (updated the exiting tests to use 'from-image')
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1342260
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1809740
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=250476
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: https://github.com/mdn/content/issues/23564

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://whatpr.org/html/8710/imagebitmap-and-animations.html" title="Last updated on Jan 16, 2023, 12:53 PM UTC (2c7dade)">/imagebitmap-and-animations.html</a>  ( <a href="https://whatpr.org/html/8710/67994b2...2c7dade/imagebitmap-and-animations.html" title="Last updated on Jan 16, 2023, 12:53 PM UTC (2c7dade)">diff</a> )

domenic commented:
What is the relationship between this and #8687?

yiyix commented:
> What is the relationship between this and #8687?

This is like the intermediate step to get to #8687. If we repurpose none to from-image and re-define none in 1 release, we are worried that some website will suffer regression. So we want to launch this first and send deprecation warning when imageOrientation: 'none' is used, then remove 'none' and finally redefine 'none' to ignore internal image orientation.

annevk reviewed (approved):
I removed the "implementations can warn" language as that would require us to acknowledge "none" as an official value, which seems like more work than is warranted.

I also changed "required" to "needed" as "required" is a conformance term.

With those changes this looks good to me. I'll let @domenic merge.

yiyix commented:
> I removed the "implementations can warn" language as that would require us to acknowledge "none" as an official value, which seems like more work than is warranted.
> 
> I also changed "required" to "needed" as "required" is a conformance term.
> 
> With those changes this looks good to me. I'll let @domenic merge.

It makes sense to me! Thank you for making these changes.
